ee/tasks/webhooks_ee.py

```diff
@@ -12,41 +12,53 @@
 
 @app.task(ignore_result=True, bind=True, max_retries=3)
 def post_event_to_webhook_ee(self: Task, event: Dict[str, Any], team_id: int, site_url: str) -> None:
-    team = Team.objects.get(pk=team_id)
-    _event = Event.objects.create(
-        event=event["event"],
-        distinct_id=event["distinct_id"],
-        properties=event["properties"],
-        team=team,
-        site_url=site_url,
-        **({"timestamp": event["timestamp"]} if event["timestamp"] else {}),
-        **({"elements": event["elements_list"]} if event["elements_list"] else {})
-    )
-
-    actions = cast(Sequence[Action], Action.objects.filter(team_id=team_id, post_to_slack=True).all())
-
     if not site_url:
         site_url = settings.SITE_URL
+    postgres_event = None
+    try:
+        team = Team.objects.select_related("organization").get(pk=team_id)
+        postgres_event = Event.objects.create(
+            event=event["event"],
+            distinct_id=event["distinct_id"],
+            properties=event["properties"],
+            team=team,
+            site_url=site_url,
+            **({"timestamp": event["timestamp"]} if event["timestamp"] else {}),
+            **({"elements": event["elements_list"]} if event["elements_list"] else {})
+        )
+        is_zapier_available = team.organization.is_feature_available("zapier")
 
-    for action in actions:
-        qs = Event.objects.filter(pk=_event.pk).query_db_by_action(action)
-        if not qs:
-            continue
-        # REST hooks
-        action.on_perform(_event)
-        # webhooks
-        if not team.slack_incoming_webhook:
-            continue
-        message_text, message_markdown = get_formatted_message(action, _event, site_url,)
-        if determine_webhook_type(team) == "slack":
-            message = {
-                "text": message_text,
-                "blocks": [{"type": "section", "text": {"type": "mrkdwn", "text": message_markdown},},],
-            }
-        else:
-            message = {
-                "text": message_markdown,
-            }
-        requests.post(team.slack_incoming_webhook, verify=False, json=message)
+        actionFilters = {"team_id": team_id}
+        if not is_zapier_available:
+            if not team.slack_incoming_webhook:
+                return  # Exit this task if neither Zapier nor webhook URL are available
```

Comment:
Bug: This will leave the created event laying around

I think we should check this condition before creating the event